### PR TITLE
add events to #r nuget

### PR DESCRIPTION
**Pull request by brettfo**
Includes events for dependency add start, successful completion, and failure.

The events are fired in `DependencyManager.Integration.fs` in the `ResolveDependencies` method, the rest is just plumbing.  Also included in this is the removal of copypasta code where the `FSharpScript` class was originally duplicated to make testing easier, but now that scripting and #r nuget are in the same place, the duplicate code can go.

### CODE REVIEW

File: src/fsharp/DependencyManager.Integration.fs [hunk 1]
```diff
@@ -75,6 +78,10 @@ type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyIn
     let nameProperty     = nameProperty.GetValue >> string
     let keyProperty      = keyProperty.GetValue >> string
 
+    let dependencyAddingEvent = new Event<_>()
+    let dependencyAddedEvent = new Event<_>()
+    let dependencyFailedEvent = new Event<_>()
```
Comment (cartermp) [inline src/fsharp/DependencyManager.Integration.fs]: `new` shouldn't be necessary here

File: src/fsharp/DependencyManager.Integration.fs [hunk 2]
```diff
@@ -91,9 +98,18 @@ type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyIn
     interface IDependencyManagerProvider with
         member __.Name     = instance |> nameProperty
         member __.Key      = instance |> keyProperty
-        member __.ResolveDependencies(scriptDir, mainScriptName, scriptName, packageManagerTextLines, tfm) =
+        member this.ResolveDependencies(scriptDir, mainScriptName, scriptName, packageManagerTextLines, tfm) =
+            let key = (this :> IDependencyManagerProvider).Key
+            let eventValue = String.Join("\n", packageManagerTextLines)
```
Comment (KevinRansom) [inline src/fsharp/DependencyManager.Integration.fs]: @brettfo,
should we fire an event per package (I.e. per packageManagers line) or one event for all packages per submission?

and perhaps not fire an event for Adding, Added and Failed that have already been fired?

File: src/fsharp/DependencyManager.Integration.fs (hunk 1 above)
Comment (brettfo) [reply src/fsharp/DependencyManager.Integration.fs]: Fixed, will push changes shortly.

File: src/fsharp/DependencyManager.Integration.fs (hunk 2 above)
Comment (brettfo) [reply src/fsharp/DependencyManager.Integration.fs]: Updated to trigger the event for each package manager line.

I also added a test that verifies we don't duplicate events.  E.g., the following code results in 2 events being fired, which means the one for `System.Collections.Immutable` wasn't repeated for the second submission:

``` F#
script.Eval("#r \"nuget:System.Collections.Immutable\"")
script.Eval("#r \"nuget:Newtonsoft.Json\"")
```
